### ISSUE TITLE
Regression in 1.16 RE comments in function calls (babel/typescript)

### ISSUE THREAD

**Issue by sgoll** [prettier/prettier#5832]
**Prettier 1.16.4**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEAbCAnOBbAFAHSgAIjcBKIgXgD4jgBfAGkJIHpWiBJABwGcBXbCyIBtACYR0GRkV4BLGAF1mUMgG4QjEBG4w50XslABDDBggB3AAqmEhlMYBuEOWM0gARhmNgA1nBgAZW4fOSgAc2QYDH44LQALGGxUAHV4hTheELA4QLsFOUcFAE9kcF5DLTDeOAwYK29w7GNkADNjVBqtACteAA8AIW8-AMDjbDgAGTC4No6ukF6+wLDw1DgARX4IeDnOuJAQjBqMMphi7kywDDldd24b2BTXGHjkAA4ABi0HiBqU7zcMoPTK1RyzLRYACO-DkWAaxiaLSQ7X2Whq2DkURiB3kEXWWx2sxR8wOMGMHmeYleyAATFposY5KhVgBhCDYZplKDQCEgfg1AAqFPsqJq9HoQA)

**Input:**
```jsx
lorem(
  () => {},
  // Ipsum
  [dolor, sit],
);
```

**Output:**
```jsx
lorem(() => {}, // Ipsum
[dolor, sit]);
```

**Expected behavior:**
The comment should not be moved to the previous line, as was the behavior in prettier@1.15.

**Furter details:**
The problem happens with the `flow` and `typescript` parsers but does not happen with the `babel`/`babel-flow` parsers. Applying prettier again to the formatted output results in this:

```jsx
lorem(() => {}, [dolor, sit]); // Ipsum
```

In other words, prettier is no longer idempotent and the comment location is entirely lost.

**Comment by sgoll** [comment]
On further investigation, this seems to be related to #5608.

**Comment by thorn0** [comment]
Fixed in version 1.19.
